### PR TITLE
AF-1674: Role-based access to branches in the Workbench

### PR DESCRIPTION
**Pull request by paulovmr**
Part of an ensemble:
* https://github.com/kiegroup/appformer/pull/665
* https://github.com/kiegroup/kie-wb-common/pull/2538
* https://github.com/kiegroup/drools-wb/pull/1095
* https://github.com/kiegroup/jbpm-designer/pull/843
* https://github.com/kiegroup/jbpm-wb/pull/1319
* https://github.com/kiegroup/optaplanner-wb/pull/330

### CODE REVIEW

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins retest this

File: kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-client/src/main/java/org/kie/workbench/common/forms/editor/client/editor/FormEditorPresenter.java [hunk 1]
```diff
@@ -296,24 +296,33 @@ public IsWidget getWidget() {
...
+                            .addCopy(this::safeCopy)
+                            .addRename(this::safeRename)
+                            .addDelete(this::safeDelete);
+                }
+                addDownloadMenuItem(fileMenuBuilder);
+
+                fileMenuBuilder
+                        .addNewTopLevelMenu(versionRecordManager.buildMenu())
+                        .addNewTopLevelMenu(alertsButtonMenuItemBuilder.build());
+                        /*.addCommand( "PREVIEW",
```
Comment (ederign) [inline kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-client/src/main/java/org/kie/workbench/common/forms/editor/client/editor/FormEditorPresenter.java]: Could you please clean this?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementPresenter.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementPresenter.java]: Fixed.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementPresenter.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.workbench.common.screens.library.client.settings.sections.branchmanagement;
+
+import java.util.ArrayList;
```
[the remark this answers is not in the document]
Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementPresenter.java]: Fixed.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementView.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<!--
+  ~ Copyright 2017 Red Hat, Inc. and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementView.html]: Fixed.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementView.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/BranchManagementView.java]: Fixed.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/RoleItemPresenter.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/RoleItemPresenter.java]: Fixed.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/RoleItemView.html [hunk 7]
```diff
@@ -0,0 +1,41 @@
+<!--
+  ~ Copyright 2017 Red Hat, Inc. and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/RoleItemView.html]: Fixed.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/RoleItemView.java [hunk 8]
```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (paulovmr) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/settings/sections/branchmanagement/RoleItemView.java]: Fixed.

**Comment by paulovmr** [comment]
HI @barboras7 , thanks for the review. As I mentioned on IRC, I fixed all things you commented and also resolved some conflicts. Based on these conflicts, it would be good if you could make a quick test on the data object editor, check if the user that doesn't have permission to save it really can't modify it (read only). Thanks!

**Comment by paulovmr** [comment]
jenkins retest this

**Pull request merged**